src/consensus/params.cpp

```diff
@@ -5,12 +5,30 @@
 #include "params.h"
 
 #include "upgrades.h"
+#include "util.h"
 
 namespace Consensus {
     bool Params::NetworkUpgradeActive(int nHeight, Consensus::UpgradeIndex idx) const {
         return NetworkUpgradeState(nHeight, *this, idx) == UPGRADE_ACTIVE;
     }
 
+    bool Params::FeatureActive(const int nHeight, const Consensus::ConsensusFeature feature) const {
```

Comment:
This FeatureActive check is the core piece of functionality that is added to permit flagging of new consensus rules.